### PR TITLE
Convenience methods in org member commander

### PR DESCRIPTION
**Pull request by ryanpbrewster**
Also fixed an edgecase where users could not remove themselves.
The true owner of an org cannot remove themself. They must delete the org.

@leogrim @Colin-Lane

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationMembershipCommander.scala [hunk 1]
```diff
@@ -116,12 +119,19 @@ class OrganizationMembershipCommanderImpl @Inject() (
           targetOpt.exists(_.role < requester.role) && requester.permissions.contains(MODIFY_MEMBERS)
 
         case OrganizationMembershipRemoveRequest(_, _, _) =>
-          targetOpt.exists(_.role < requester.role) &&
-            requester.permissions.contains(REMOVE_MEMBERS)
+          val selfRemove = targetOpt.exists(t => t.userId == requester.userId && t.userId != org.ownerId)
+          val otherRemove = targetOpt.exists(_.role < requester.role) && requester.permissions.contains(REMOVE_MEMBERS)
+          selfRemove || otherRemove
```
Comment (Colin-Lane) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationMembershipCommander.scala]: this doesn't take advantage of `short-circuit evaluation`. Not really necessary to change though.

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationMembershipCommander.scala]: As mentioned somewhere else, can we use verbs for boolean when it makes sense?
e.g. here `canLeave` and `canRemoveOther`

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationMembershipCommander.scala [hunk 2]
```diff
@@ -116,12 +119,19 @@ class OrganizationMembershipCommanderImpl @Inject() (
...
       }
     }
   }
 
+  def validateRequests(requests: Seq[OrganizationMembershipRequest]): Map[OrganizationMembershipRequest, Boolean] = {
+    db.readOnlyReplica { implicit session =>
+      requests.map(r => r -> validRequest(r)).toMap
+    }
+  }
```
Comment (Colin-Lane) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationMembershipCommander.scala:135]: `requests.filter(validRequest(_)).map(_ -> true).toMap.withDefaultValue(false)`
Saves memory, and if you missed any `OrganizationMembershipRequest` they will default to invalid.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationMembershipCommander.scala (hunk 1 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationMembershipCommander.scala]: Do we have any convention to distinguish values from functions? I had fallen into the habit of "functions as verbs" and "values as nouns", but I see your point regarding Booleans.

**Pull request merged**